From 1e1c3cc0761879e8a2e4d4a884dacc555bbe7fa0 Mon Sep 17 00:00:00 2001
From: James Le Cuirot <chewi@gentoo.org>
Date: Fri, 9 Mar 2018 23:06:43 +0000
Subject: [PATCH 2/2] build: Fix misleading TIRPC result in configure summary

Requesting ipv6-default even if you explicitly disable libtirpc will
then implicitly enable libtirpc because that is required. That is fine
but the configure summary should not then show TIRPC as disabled when
it is not.

The result has also been made clearer by stating that TIRPC is
"missing" when it has been tried but not found.

BUG: 1553938
Change-Id: I945bd6859aaf3defa682b0d05ee34a9827b9c45f
Signed-off-by: James Le Cuirot <chewi@gentoo.org>
---
 configure.ac | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index d42179182..9571dac76 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1097,16 +1097,15 @@ AC_SUBST(GF_DISTRIBUTION)
 GF_HOST_OS=""
 GF_LDFLAGS="-rdynamic"
 
-DISABLE_LIBTIRPC=no
 dnl include tirpc for IPv6 builds
 if test "x$with_libtirpc" = "xyes" || test "x$with_ipv6_default" = "xyes" ; then
     PKG_CHECK_MODULES([TIRPC], [libtirpc],
-       [GF_CFLAGS="$GF_CFLAGS $TIRPC_CFLAGS"; GF_LDFLAGS="$GF_LDFLAGS $TIRPC_LIBS";],
-       [DISABLE_LIBTIRPC=yes])
+       [with_libtirpc=yes; GF_CFLAGS="$GF_CFLAGS $TIRPC_CFLAGS"; GF_LDFLAGS="$GF_LDFLAGS $TIRPC_LIBS";],
+       [with_libtirpc=missing])
 fi
 
-if test "x$DISABLE_LIBTIRPC" = "xyes" ; then
-    with_libtirpc=no; with_ipv6_default=no
+if test "x$with_libtirpc" = "xmissing" ; then
+    with_ipv6_default=no
     AC_CHECK_HEADERS([rpc/rpc.h],[
         AC_MSG_WARN([
             ---------------------------------------------------------------------------------
-- 
2.16.1

